plugins/inputs/gnmi/gnmi.go

```diff
@@ -437,11 +545,43 @@ func (c *GNMI) handleTelemetryField(update *gnmiLib.Update, tags map[string]stri
 			}
 		}
 	}
-	return aliasPath, fields
+
+	return aliasPath, fields, false
+}
+
+func decodeTag(update *gnmiLib.Update) string {
+	switch val := update.Val.Value.(type) {
+	case *gnmiLib.TypedValue_AsciiVal:
+		return val.AsciiVal
+	case *gnmiLib.TypedValue_BoolVal:
+		if val.BoolVal {
+			return "true"
+		}
+		return "false"
+	case *gnmiLib.TypedValue_BytesVal:
+		return string(val.BytesVal)
+	case *gnmiLib.TypedValue_DecimalVal:
+		return strconv.FormatFloat(float64(val.DecimalVal.Digits)/math.Pow(10,
+			float64(val.DecimalVal.Precision)), 'f',
+			-1, 64)
+	case *gnmiLib.TypedValue_FloatVal:
+		strconv.FormatFloat(float64(val.FloatVal), 'f', -1, 32)
+	case *gnmiLib.TypedValue_IntVal:
+		return strconv.FormatInt(val.IntVal, 10)
+	case *gnmiLib.TypedValue_StringVal:
+		return val.StringVal
+	case *gnmiLib.TypedValue_UintVal:
+		return strconv.FormatUint(val.UintVal, 10)
+	case *gnmiLib.TypedValue_JsonIetfVal:
+		return string(val.JsonIetfVal)
+	case *gnmiLib.TypedValue_JsonVal:
+		return string(val.JsonVal)
+	}
+	return ""
 }
```

Comment:
Isn't this similar to the [internal `ToString` function](https://github.com/influxdata/telegraf/blob/f7aab29381798bc27d877bff238643246fa719a9/internal/type_conversions.go#L8)?